test/Concurrency/Runtime/async_task_locals_prevent_illegal_use.swift

```diff
@@ -0,0 +1,42 @@
+// RUN: %target-fail-simple-swift(-Xfrontend -enable-experimental-concurrency -parse-as-library %import-libdispatch) 2>&1 | %FileCheck %s
+//
+// This test is intended to *crash*, so we're using target-fail-simple-swift
+// which expects the exit code of the program to be non-zero;
+// We then check stderr for the expected error message using filecheck as usual.
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: libdispatch
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+enum TL {
+  @TaskLocal
+  static var number: Int  = 0
+}
+
+// ==== ------------------------------------------------------------------------
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+func bindAroundGroupSpawn() async {
+  await TL.$number.withValue(1111) { // ok
+    await withTaskGroup(of: Int.self) { group in
+
+      // CHECK: error: task-local: detected illegal task-local value binding at {{.*}}illegal_use.swift:[[# @LINE + 1]]
+      await TL.$number.withValue(2222) { // bad!
+        print("Survived, inside withValue!") // CHECK-NOT: Survived, inside withValue!
+        group.spawn {
+          0 // don't actually perform the read, it would be unsafe.
+        }
+      }
+
+      print("Survived the illegal call!") // CHECK-NOT: Survived the illegal call!
```

Comment:
so i made this cool test to check that we even crash properly, it needed the new lit command though; I could not figure out how to use `expectCrash` to achieve the same good result checking line numbers etc.

Looks ok, mind if we keep this? It matters a lot to test crashes look good IMO